Google.Api.Generator.Tests/ProtoTests/Basic/gapic_metadata.json

```diff
@@ -0,0 +1,24 @@
+{
+  "schema": "1.0",
+  "comment": "This file maps proto services/RPCs to the corresponding library clients/methods",
+  "language": "csharp",
+  "protoPackage": "testing.basic",
+  "libraryPackage": "Testing.Basic",
+  "services": {
+    "Basic": {
+      "clients": {
+        "grpc": {
+          "libraryClient": "Testing.Basic.BasicClientImpl",

```

Comment:
Should this be the partial class or just "BasicClient"? Genuinely curious what C# conventions recommend.

Also, the `Testing.Basic` package probably isn't needed here.

Reply:
My line of thinking was that if a consumer of this file `Ctrl+T`s `BasicClientImpl` and finds `AMethod` function they will be at the implementation, whereas same for `BasicClient` will place them at a virtual function throwing `NotImplementedException` which seems like a worse experience. Can see an argument for opposite from theoretical standpoint though.

Agree that repeating of namespace is not needed for the current version of the spec, fixed.

Reply:
I'd missed this before - I'd prefer it to be `BasicClient`, as that's where the method is first declared, and it's what most users will actually use. (We only make the `Impl` class public so that folks *can* construct it directly and use it for faking purposes if they absolutely need to.) I suspect most users aren't even aware that the `Impl` classes are available.

Reply:
👍